File: CPP/twoSum.cpp
#include<iostream>
#include<map>
#include<vector>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        vector<int> res;
        for(int  i=0; i<nums.size();i++){
            if (m.find(nums[i]) != m.end()){
                res.push_back(m[nums[i]]);
                res.push_back(i);
                return res;
            }
            else{
                m.insert(pair <int,int>(target-nums[i],i));
            }
        }


    }
int main()
{
  vector<int> nums = {3,2,4};
  int target = 6;
  vector<int> res = twoSum(nums,target);
  for (int i: res)
    cout<< i << endl;
  return 0;
}

File: Python/twoSum.py
def twoSum(nums,target):
    temp={}
    for i, element in enumerate(nums):
        if element in temp.keys():
            return [temp[element],i]
        else:
            temp[target - element]=i
    print(temp)

nums = [2,7,11,15]
target = 9
print(twoSum(nums,target))
